app/javascript/src/components/editor/CodeMirror.svelte

```diff
@@ -165,6 +166,34 @@
     }
   }
 
+  function autocompleteFormatting(view, transaction) {
+    if (transaction.transactions.every(tr => ["input.complete"].includes(tr.annotation(Transaction.userEvent)))) {
+      indentMultilineInserts(view, transaction)
+      insertSemicolon(view, transaction)
```

Comment:
Could this maybe be optional? 😁

Reply:
Just don't place the semicolon in any situation. Make it so I have to add it myself (because I may have the reflex of doing so automatically).

I agree on that such a setting may clutter the settings section. But maybe its time to make that into a separate dialog (a.k.a. on another PR :D)